gems/pending/appliance_console/prompts.rb

```diff
@@ -85,6 +85,21 @@ def ask_for_ip_or_hostname_or_none(prompt, default = nil)
       ask_for_ip(prompt, default, validation, "a valid Hostname or IP Address.").gsub(/^'?NONE'?$/i, "")
     end
 
+    def ask_for_schedule_frequency(prompt = "Select schedule frequency: hourly, daily, weekly or monthly", default = "hourly")
+      validation = ->(h) { h.empty? || h =~ /^hourly$/i || h =~ /^daily$/i || h =~ /^weekly$/i || h =~ /^monthly$/i }
+      just_ask(prompt, default, validation, "hourly, daily, weekly or monthly")
+    end
+
+    def ask_for_hour_number(prompt = "Select hour(0-23)", default = "0")
+      validation = ->(h) { h.empty? || h =~ /^(#{(0..23).to_a * '|'})$/ }
```

Comment:
Yes but there is a catch. I will need to test for empty because the empty string matches 0 and "0", depending if the range is an integer or a string.


```ruby
>> (0..6).include?("".to_i)
=> true

>> "0..6".include?("")
=> true
```

So the solution would be:

```ruby
validation = ->(h) { ! h.empty? && (0..23).include?(h.to_i) }
```